test/e2e/e2e_test.go

```diff
@@ -824,12 +827,40 @@ func createFileDoesntExistCmd(testFileDir string, testFile string) string {
 	return fmt.Sprintf("[ ! -e %s ]", testFilePath)
 }
 
+func savePodLogs(client clientset.Interface, dir string, pods []v1.Pod) {
+	for _, pod := range pods {
+		logs, err := framework.GetPodLogs(client, pod.Namespace, pod.Name, "")
+		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			continue
+		}
+		logPath := filepath.Join(dir, fmt.Sprintf("pod_%s_%s_%s.log", pod.Namespace, pod.Name, pod.UID))
+		file, err := os.Create(logPath)
+		if err != nil {
+			continue
+		}
+		defer file.Close()
+		file.WriteString(logs)
+	}
+}
+
 func deleteProvisionerDaemonset(config *localTestConfig) {
 	ds, err := config.client.ExtensionsV1beta1().DaemonSets(config.ns).Get(daemonSetName, metav1.GetOptions{})
 	if ds == nil {
 		return
 	}
 
+	// save pod logs for further debugging
+	if framework.TestContext.ReportDir != "" {
+		podList, err := config.client.CoreV1().Pods(config.ns).List(metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("app=%s", daemonSetName),
+		})
+		if err != nil {
+			framework.Failf("could not get the pod list: %v", err)
+		}
+		savePodLogs(config.client, framework.TestContext.ReportDir, podList.Items)
```

Comment:
alternatively we can deploy pod logs collector (e.g. fluentd), but this is much simpler way